mitodl/odl-video-service
# Video Analytics Backend

#### What are the relevant tickets?
Partially implements #402 

#### What's this PR do?
1. Adds backend utility functions to query Google Analytics for event stats for a particular video
2. Adds an API endpoint to expose analytics data.

#### How should this be manually tested?
1. In `<ROOT>/.env, set...
   a. GA_DIMENSION_CAMERA=dimension1
   b. GA_VIEW_ID=<the ID of an analytics view that provides video event data>
   c. GA_KEYFILE=<path to the keyfile for Google service account that has access to the analytics view>

2. After starting the app, go to `http://localhost:8089/api/v0/videos/<SOME VIDEO KEY>/analytics/`. You should see the DjangoRestFramework view, w/ analytics data corresponding to that video.

#### Known Issues
1. video keys that begin with a letter, rather than a number, appear to have that letter capitalized when event labels are sent to google analytics. This means that queries for videos with leading-letter keys fail to find matching events. >:/   I do not know yet where this capitalization is happening.

#### Suggested Reviewers
@mbertrand perhaps?
